### PR TITLE
Update resource types to use hardened images

### LINKED ISSUE #4384: Replace docker resource types with harden images for pipelines (core)

### PR DESCRIPTION
**Pull request by svenaas**
Fixes #4384 

## Changes proposed in this pull request:
- Update pipelines to use our current hardened images
- Clean up on hook notifications

## security considerations
Replace CI resource types with hardened image

### CODE REVIEW

File: ci/pipeline-production.yml [hunk 1]
```diff
@@ -550,15 +567,26 @@ resource_types:
   - name: github-release
     type: registry-image
     source:
-      repository: concourse/github-release-resource
+      aws_access_key_id: ((ecr_aws_key))
+      aws_secret_access_key: ((ecr_aws_secret))
+      repository: github-pr-resource
```
Comment (apburnes) [inline ci/pipeline-production.yml:572]: This is hardened resource type is not the same as `concourse/github-release-resource`. We can see if we can add it to common pipelines.

File: ci/pipeline-production.yml [hunk 2]
```diff
@@ -381,9 +403,11 @@ jobs:
               VULN_THRESHOLD: 8 # errors on critical or high
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:406]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 3]
```diff
@@ -394,9 +418,11 @@ jobs:
             icon_url: ((slack-icon-url))
     on_success:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: success}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:421]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml (hunk 1 above)
Comment (svenaas) [reply ci/pipeline-production.yml:572]: My mistake. Would the common-piplines git-resource cover what we need?

File: ci/pipeline-production.yml [hunk 4]
```diff
@@ -56,9 +56,11 @@ jobs:
         passed: [set-pipeline]
         trigger: true
         params: {depth: 1}
-      - put: gh-status
-        inputs: [src]
-        params: {state: pending}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:59]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 5]
```diff
@@ -104,9 +106,11 @@ jobs:
           APP_ENV: ((deploy-env))
         on_failure:
           in_parallel:
-            - put: gh-status
-              inputs: [src]
-              params: {state: failure}
+            - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:109]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 6]
```diff
@@ -118,9 +122,11 @@ jobs:
 
         on_failure:
           in_parallel:
-            - put: gh-status
-              inputs: [src]
-              params: {state: failure}
+            - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:125]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 7]
```diff
@@ -132,9 +138,11 @@ jobs:
 
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:141]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 8]
```diff
@@ -145,9 +153,11 @@ jobs:
             icon_url: ((slack-icon-url))
     on_success:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: success}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:156]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 9]
```diff
@@ -165,9 +175,11 @@ jobs:
         trigger: true
         params: {depth: 1}
       - get: node
-      - put: gh-status
-        inputs: [src]
-        params: {state: pending}
+      - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:178]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 10]
```diff
@@ -197,9 +209,11 @@ jobs:
 
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:212]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 11]
```diff
@@ -211,9 +225,11 @@ jobs:
 
     on_success:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: success}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:228]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 12]
```diff
@@ -230,9 +246,11 @@ jobs:
         passed: [set-pipeline]
         trigger: true
         params: {depth: 1}
-      - put: gh-status
-        inputs: [src]
-        params: {state: pending}
+      - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:249]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 13]
```diff
@@ -253,9 +271,11 @@ jobs:
               CF_APP_NAME: pages-queues-ui-((deploy-env))
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:274]: Remove because this is not run from a PR.

File: ci/pipeline-production.yml [hunk 14]
```diff
@@ -266,9 +286,11 @@ jobs:
             icon_url: ((slack-icon-url))
     on_success:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: success}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-production.yml:289]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 15]
```diff
@@ -161,9 +161,11 @@ jobs:
         passed: [set-pipeline]
         trigger: true
         params: {depth: 1}
-      - put: gh-status
-        inputs: [src]
-        params: {state: pending}
+      - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:164]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 16]
```diff
@@ -209,9 +211,11 @@ jobs:
           APP_ENV: ((deploy-env))
         on_failure:
           in_parallel:
-            - put: gh-status
-              inputs: [src]
-              params: {state: failure}
+            - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:214]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 17]
```diff
@@ -223,9 +227,11 @@ jobs:
 
         on_failure:
           in_parallel:
-            - put: gh-status
-              inputs: [src]
-              params: {state: failure}
+            - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:230]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 18]
```diff
@@ -237,9 +243,11 @@ jobs:
 
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:246]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 19]
```diff
@@ -250,9 +258,11 @@ jobs:
             icon_url: ((slack-icon-url))
     on_success:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: success}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:261]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 20]
```diff
@@ -270,9 +280,11 @@ jobs:
         trigger: true
         params: {depth: 1}
       - get: node
-      - put: gh-status
-        inputs: [src]
-        params: {state: pending}
+      - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:283]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 21]
```diff
@@ -302,9 +314,11 @@ jobs:
 
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:317]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 22]
```diff
@@ -316,9 +330,11 @@ jobs:
 
     on_success:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: success}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:333]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 23]
```diff
@@ -335,9 +351,11 @@ jobs:
         passed: [set-pipeline]
         trigger: true
         params: {depth: 1}
-      - put: gh-status
-        inputs: [src]
-        params: {state: pending}
+      - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:354]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 24]
```diff
@@ -358,9 +376,11 @@ jobs:
               CF_APP_NAME: pages-queues-ui-((deploy-env))
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:379]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 25]
```diff
@@ -371,9 +391,11 @@ jobs:
             icon_url: ((slack-icon-url))
     on_success:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: success}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:394]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 26]
```diff
@@ -455,9 +477,11 @@ jobs:
               VULN_THRESHOLD: 8 # errors on critical or high
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:480]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 27]
```diff
@@ -468,9 +492,11 @@ jobs:
             icon_url: ((slack-icon-url))
     on_success:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: success}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:495]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 28]
```diff
@@ -536,9 +562,11 @@ jobs:
         - deploy-queues-ui-pages
       - get: node
       - get: cf-image
-      - put: gh-status
-        inputs: [src]
-        params: {state: pending}
+      - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:565]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 29]
```diff
@@ -565,9 +593,11 @@ jobs:
           - put: s3
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:596]: Remove because this is not run from a PR.

File: ci/pipeline-staging.yml [hunk 30]
```diff
@@ -578,9 +608,11 @@ jobs:
             icon_url: ((slack-icon-url))
     on_success:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: success}
+        - put: pr-((git-branch))
```
Comment (apburnes) [inline ci/pipeline-staging.yml:611]: Remove because this is not run from a PR.

**Review by apburnes — CHANGES_REQUESTED** [review]
Make sure to also add the hardened git resource for the staging and production pipelines.

File: ci/pipeline-production.yml (hunk 1 above)
Comment (drewbo) [reply ci/pipeline-production.yml:572]: No, there is a specific release resource on the list though, I can't remember if it has been hardened yet

**Comment by svenaas** [comment]
@apburnes Thanks for that review — that was a specific detail I was unsure of. I believe I've addressed all of the `pr-((git-branch))` instances that needed to be removed.

**Comment by svenaas** [comment]
> @apburnes Thanks for that review — that was a specific detail I was unsure of. I believe I've addressed all of the `pr-((git-branch))` instances that needed to be removed.

(There are more in the dev pipeline, but in a PR context, I believe.)

**Comment by svenaas** [comment]
> Make sure to also add the hardened git resource for the staging and production pipelines.

Not sure which one you mean. cogito is removed and github-pr-resource is added in those pipelines. Do we need git-resource also?

File: ci/pipeline-production.yml (hunk 1 above)
Comment (drewbo) [reply ci/pipeline-production.yml:497]: https://github.com/cloud-gov/product/issues/2618, not hardened yet but tracked, we should use the existing one for now

Comment (svenaas) [reply ci/pipeline-production.yml:497]: @drewbo Isn't that the one I'm using? That Andrew said isn't right here?

Comment (drewbo) [reply ci/pipeline-production.yml:497]: we should use the unhardened `concourse/github-release-resource` here (so no change from the original file). It is scheduled to be hardened (linked ticket) and we can replace it then

Comment (svenaas) [reply ci/pipeline-production.yml:497]: Got it. Thanks.

**Comment by svenaas** [comment]
I think I've addressed every comment except for the following, about which my open question is below:

> > Make sure to also add the hardened git resource for the staging and production pipelines.
> 
> Not sure which one you mean. cogito is removed and github-pr-resource is added in those pipelines. Do we need git-resource also?

**Comment by svenaas** [comment]
Thanks for swinging in on this!

**Comment by svenaas** [comment]
Noticing I didn't include "(#4384)" in the original commit message, but maybe we can let that be.

**Comment by svenaas** [comment]
> Noticing I didn't include "(#4384)" in the original commit message

That would be because this PR is #4383 and I would have had to time travel. Which `git rebase -i` allows...

**Review by drewbo — APPROVED** [review]

**Pull request merged**